### PR TITLE
Explicitly define parameters for failingTimeout

### PR DESCRIPTION
**Pull request by eugene-davis**
Provides variants of the failingTimeout function that handle different argument sets, allowing this function to be used in either scripted  pipelines or declarative pipelines.

### CODE REVIEW

File: vars/failingTimeout.groovy [hunk 1]
```diff
@@ -28,4 +39,20 @@ def call(Map conf, Closure body) {
             throw err
         }
```
Comment (martinm82) [inline vars/failingTimeout.groovy]: Wrong indentation

File: vars/failingTimeout.groovy [hunk 2]
```diff
@@ -28,4 +39,20 @@ def call(Map conf, Closure body) {
             throw err
```
Comment (martinm82) [inline vars/failingTimeout.groovy]: Wrong indentation

File: vars/failingTimeout.groovy [hunk 3]
```diff
@@ -13,10 +13,21 @@
...
+        activity = args.activity
+    }
+
+    if (args.containsKey('unit')) {
+        unit = args.unit
+    }
+
     try {
-        timeout(conf) {
+        timeout(time: args.time, activity: activity, unit: unit) {
```
Comment (martinm82) [inline vars/failingTimeout.groovy:30]: Where did you find the activity argument? Based on https://jenkins.io/doc/book/pipeline/syntax/ I don't see it

**Review by martinm82 — CHANGES_REQUESTED** [review]

**Review by martinm82 — APPROVED** [review]

File: vars/failingTimeout.groovy (hunk 3 above)
Comment (eugene-davis) [reply vars/failingTimeout.groovy:30]: It's here https://jenkins.io/doc/pipeline/steps/workflow-basic-steps/#timeout-enforce-time-limit

**Pull request merged**